src/util/allocate_objects.h

```diff
@@ -81,7 +81,7 @@ class allocate_objectst
 private:
   const irep_idt &symbol_mode;
   const source_locationt &source_location;
```

Comment:
And this one as well - indeed the test should actually break here, because the temporary will also go out of scope.